### PR TITLE
Don't display tooltip when target is under cta layer 🐛

### LINKED ISSUE #19286: [amp-story] Clickable targets inside amp-story-cta-layer are displaying a tooltip

### PR DESCRIPTION
**Pull request by Enriqe**
Fixes #19286

### CODE REVIEW

**Comment by Enriqe** [comment]
Updated the PR to avoid traversing the tree twice. @newmuis @gmajoulet PTAL so we can make it to the canary cut 👍

File: extensions/amp-story/1.0/page-advancement.js [hunk 1]
```diff
@@ -426,6 +426,21 @@ class ManualAdvancement extends AdvancementConfig {
     });
   }
 
+  /**
+   * Checks if current element is valid to trigger a tooltip.
+   * @param {!Event} event
+   */
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/page-advancement.js]:
```
@return {boolean}
@private 
```

File: extensions/amp-story/1.0/page-advancement.js [hunk 2]
```diff
@@ -426,6 +426,21 @@ class ManualAdvancement extends AdvancementConfig {
...
+  isTooltipValidDescendant_(event) {
+    let valid = true;
+    return !!closest(dev().assertElement(event.target), el => {
+      if (el.tagName.toLowerCase() == 'amp-story-cta-layer') {
+        valid = false;
+        return false;
+      }
+      return el.tagName.toLowerCase() == 'amp-story-page' && valid;
+    });
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/page-advancement.js]: Maybe consider passing `this.element_` as a third parameter to stop traversing the DOM once we reach `<amp-story>`?

File: extensions/amp-story/1.0/page-advancement.js [hunk 3]
```diff
@@ -426,6 +426,21 @@ class ManualAdvancement extends AdvancementConfig {
     });
   }
 
+  /**
+   * Checks if current element is valid to trigger a tooltip.
+   * @param {!Event} event
+   */
+  isTooltipValidDescendant_(event) {
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/page-advancement.js]: How about a `canShowTooltip_` instead? With a comment saying that it has to be descendant of `amp-story-page` but not `amp-story-cta-layer`?

File: extensions/amp-story/1.0/page-advancement.js (hunk 2 above)
Comment (Enriqe) [reply extensions/amp-story/1.0/page-advancement.js]: sure, it's important to note `this.element_` could also be an `AmpStoryPage` but that is probably fine.

File: extensions/amp-story/1.0/page-advancement.js (hunk 1 above)
Comment (Enriqe) [reply extensions/amp-story/1.0/page-advancement.js]: Done.

File: extensions/amp-story/1.0/page-advancement.js (hunk 3 above)
Comment (Enriqe) [reply extensions/amp-story/1.0/page-advancement.js]: Done.

**Comment by Enriqe** [comment]
ready to merge @gmajoulet

**Pull request merged**